crates/database/src/ethflow_orders.rs

```diff
@@ -130,12 +133,8 @@ AND o.partially_fillable = false
 AND t.order_uid is null
 AND eo.valid_to < $1
 AND o.sell_amount = oq.sell_amount
-AND (
-    CASE
-        WHEN oq.buy_amount = 0 THEN TRUE
-        ELSE (1.0 - o.buy_amount / oq.buy_amount) >= $3
-    END
-)
+AND oq.buy_amount > 0
+AND (1.0 - o.buy_amount / oq.buy_amount) >= $3
```

Comment:
```suggestion
AND (1.0 - o.buy_amount / GREATEST(oq.buy_amount,1)) >= $3
```

Reply:
Ah ok, you implicitly assume `buy_amount` is integer and not f64, all good